tide/wall/WallRenderContext.h

```diff
@@ -0,0 +1,80 @@
+/*      disclaimer.                                                  */
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#ifndef WALLRENDERCONTEXT_H
+#define WALLRENDERCONTEXT_H
+
+#include "types.h"
+
+#include <QQmlContext>
+#include <QQmlEngine>
+
+/**
+ * Context for rendering Qml items in a wall window.
+ */
+struct WallRenderContext
+{
+    QQmlEngine& engine;     //< The Qml engine
+    DataProvider& provider; //< The provider of data for Tile textures
+    QSize wallSize;         //< The total size of the wall
+    QRect screenRect;       //< The area of the wall visible in this window
+    deflect::View view;     //< The view to use for stereo contents
+
+    WallRenderContext(QQmlEngine& engine_, DataProvider& provider_,
+                      const QSize& wallSize_, const QRect& screenRect_,
+                      deflect::View view_)
+        : engine{engine_}
+        , provider{provider_}
+        , wallSize{wallSize_}
+        , screenRect{screenRect_}
+        , view{view_}
+    {
+    }
+
+    WallRenderContext(const WallRenderContext& other)
```

Comment:
= default woudn't work?

Reply:
Good point. Strangely it seems this constructor is not even required, but the first one still is. It's a bit confusing what rules the compiler applies here, and I think something changes again with C++14. I'm going to just remove it since it works without it.